File: Lab.Models/ResponseAPI.cs
using System;

namespace Lab.Models
{
    public class ResponseAPI<T> where T: new()
    {
        public int? Status { get; set; }
        public bool? Success { get; set; }
        public string Message { get; set; } = "Phản hồi không xác định";
        public string HtmlWithValidate { get; set; } = string.Empty;
        public T? Data { get; set; } = new T();
    }
}

File: vue-web/src/models/ResponseAPI.js
class ResponseAPI {
  constructor(
    callBackResult = {
      status: 500,
      success: false,
      message: "Phản hồi không xác định",
      data: null,
    }
  ) {
    console.log(callBackResult);

    this.status = callBackResult?.status || 500; // Mã trạng thái của phản hồi (status code)
    this.success = callBackResult?.success || false; // Trạng thái thành công/chưa thành công
    this.message = callBackResult?.message || "Phản hồi không xác định"; // Thông báo phản hồi từ backend
    this.data = callBackResult?.data || null; // Payload dữ liệu trả về từ backend
  }

  static empty() {
    return new ResponseAPI({
      status: null,
      success: false,
      message: "Phản hồi rỗng",
      data: null,
    });
  }
}

export default ResponseAPI;
